src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -81,41 +81,54 @@ public static Task CopyToEndAsync(this IPipeReader input, Stream stream)
         {
             try
             {
-                await input.CopyToAsync(stream, bufferSize, cancellationToken);
+                // TODO: Use bufferSize argument
+                while (!cancellationToken.IsCancellationRequested)
```

Comment:
Also, should input.Complete(TCE) be called if a cancellation is requested during the copy?